lists/transaction-types/README.md

```diff
@@ -0,0 +1,75 @@
+Transaction Type List
+=====================
+
+This is a list of existing, reserved and tentative
+[EIP-2718](https://eips.ethereum.org/EIPS/eip-2718) Transaction Types.
+
+Transaction Types
+-----------------
+
+See the sections below for additional notes on Version Bytes marked
+Reserved or Tentative.
+
+| Version Byte | Specs or Purpose |
+|--------------|------------------|
+| 0x00  | Reserved: indicates legacy (untyped) trancactions |
+| 0x01  | [EIP-2930](https://eips.ethereum.org/EIPS/eip-2930) *(available in Berlin)* |
+| 0x02  | [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) *(available in London)* |
+| 0x03  | Tentative: prevents collision with [EIP-3074](https://eips.ethereum.org/EIPS/eip-3074) |
+| 0x18  | Reserved: prevents collision with [EIP-191](https://eips.ethereum.org/EIPS/eip-191) |
+| 0x80 - 0xff  | Invalid; collides with the initial byte of valid RLP encoded transactions |
```

Comment:
I think the actual collision range is *much* smaller.  Legacy transactions are always a list, so that limits to [0xc0, 0xff].  Also, there is a shortest possible length legacy transaction which I think is a 0 ETH transfer to address 0 at 0 gas price from a constructed signature of 0 (which means from a random address).  I suspect that in reality this is shorter than is practically possible, but it gives us a lower bound of `cb` for the leading byte:
`0xcb8080825208808080808080`

For the upper bound, we know that it is not possible in the real world to submit a transaction that is 2^56 bytes long, so at the least we can free up `0xff`.  I don't know what the last constraint on transaction length would be, but at the least we could use current gas cost per calldata byte times the current block gas limit and then multiply that result by something like 100 or 1000 to get a "reasonable" upper bound (which will probably free up several more version bytes off the end).

Reply:
I was thinking the same, but the EIP seemed to imply that was the reason. I can update that `0xc0 - 0xff` are invalid and `0x80 - 0xbf` are “just reserved”.

I don’t think we should take practical size into account. Freeing up `0xff` (and ilk) seems to provide very little gain.

Reply:
At the very least I think we should only reserve 0xc0-0xff.  I see no reason to mark 0x80-0xbf as reserved.  I can appreciate the argument against freeing up the last few bytes, at least until we need them.

Reply:
I was basing this mostly on the EIP-2718, "TransactionType is a positive unsigned 8-bit number between 0 and 0x7f that represents the type of the transcation". So, there should probably be some mention of `0x80-0xc0`.

Reply:
That constraint just limits 2718 transactions to 0-7f.  It doesn't limit signature prefixes to that range.  Maybe we can mention that in this document?  0x80-0xff can be used for non-2718 transactions?

Reply:
Maybe mark `0x80-0xbf` as "Preferred; as `0x00-0x7f` are possible EIP-2718 transaction envelope  types"?

Reply:
What does "Preferred" mean?  Will it be explained in a footnote in the document?

Reply:
Might be good to make it clear what they are "preferred" for?